controllers/kyma_controller.go

```diff
@@ -94,13 +98,23 @@ func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{}, client.IgnoreNotFound(err) //nolint:wrapcheck
 	}
 
+	if kyma.Spec.Sync.Enabled {
+		var err error
```

Comment:
this `err` can be declared implicitly

Reply:
aha, got it. In this case, I think maybe it's better modify the `ctx` inside `InitializeSyncContext` directly instead of return it. But it's just an alternative, let's keep it this way.